### ISSUE TITLE
[🐛 Bug]: accessibility_id appium locator returns "InvalidArgumentException: Message: Unable to determine element locating strategy for accessibility id"

### ISSUE THREAD

**Issue by kingages** [SeleniumHQ/selenium#10140]
### What happened?

Upgrade from Grid 3 to Grid 4 does not support accessibility_id appium locator.
Setup - **does not work**:
Selenium Grid 4.1.0
Appium 1.22.1

If appium tries to identify acessibility_id it returns error `InvalidArgumentException: Message: Unable to determine element locating strategy for accessibility id`

This issue appears after upgrade from Selenium-Grid-v3.141.59 to Selenium Grid 4.1.0.

Code example
```
On Page
    Landscape
    AppiumLibrary.Wait Until Element Is Visible  accessibility_id=login_title

### How can we reproduce the issue?

```
1. Use Appium 1.22.1 with Selenium Grid 4.1.0
2. Have a iOS app with locator accessibility_id for an element
3. Run automated test that uses accessibility_id locator for the element.
```


### Relevant log output

```shell
Start / End / Elapsed:	20211213 14:27:37.453 / 20211213 14:27:37.909 / 00:00:00.456
14:27:37.453	DEBUG	POST http://192.168.65.2:4444/wd/hub/session/341fbc49-861e-4a13-8b39-5f65dc3a1998/elements {"using": "accessibility id", "value": "login_title"}	
14:27:37.463	DEBUG	http://192.168.65.2:4444 "POST /wd/hub/session/341fbc49-861e-4a13-8b39-5f65dc3a1998/elements HTTP/1.1" 400 159	
14:27:37.464	DEBUG	Finished Request	
00:00:00.441KEYWORD AppiumLibrary . Capture Page Screenshot
14:27:37.908	FAIL	InvalidArgumentException: Message: Unable to determine element locating strategy for accessibility id
14:27:37.908	DEBUG	Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "/usr/local/lib/python3.7/dist-packages/AppiumLibrary/keywords/keywordgroup.py", line 21, in _run_on_failure_decorator
    raise err
  File "/usr/local/lib/python3.7/dist-packages/AppiumLibrary/keywords/keywordgroup.py", line 16, in _run_on_failure_decorator
    return method(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/AppiumLibrary/keywords/_waiting.py", line 28, in wait_until_element_is_visible
    self._wait_until_no_error(timeout, check_visibility)
  File "/usr/local/lib/python3.7/dist-packages/AppiumLibrary/keywords/_waiting.py", line 128, in _wait_until_no_error
    timeout_error = wait_func(*args)
  File "/usr/local/lib/python3.7/dist-packages/AppiumLibrary/keywords/_waiting.py", line 21, in check_visibility
    visible = self._is_visible(locator)
  File "/usr/local/lib/python3.7/dist-packages/AppiumLibrary/keywords/_element.py", line 624, in _is_visible
    element = self._element_find(locator, True, False)
  File "/usr/local/lib/python3.7/dist-packages/AppiumLibrary/keywords/_element.py", line 574, in _element_find
    elements = self._element_finder.find(application, _locator, tag)
  File "/usr/local/lib/python3.7/dist-packages/AppiumLibrary/locators/elementfinder.py", line 36, in find
    return strategy(browser, criteria, tag, constraints)
  File "/usr/local/lib/python3.7/dist-packages/AppiumLibrary/locators/elementfinder.py", line 95, in _find_element_by_accessibility_id
    elements = browser.find_elements_by_accessibility_id(criteria)
  File "/usr/local/lib/python3.7/dist-packages/appium/webdriver/extensions/search_context/mobile.py", line 63, in find_elements_by_accessibility_id
    return self.find_elements(by=MobileBy.ACCESSIBILITY_ID, value=accessibility_id)
  File "/usr/local/lib/python3.7/dist-packages/appium/webdriver/webdriver.py", line 444, in find_elements
    return self.execute(RemoteCommand.FIND_ELEMENTS, {'using': by, 'value': value})['value'] or []
  File "/usr/local/lib/python3.7/dist-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python3.7/dist-packages/appium/webdriver/errorhandler.py", line 31, in check_response
    raise wde
  File "/usr/local/lib/python3.7/dist-packages/appium/webdriver/errorhandler.py", line 26, in check_response
    super().check_response(response)
  File "/usr/local/lib/python3.7/dist-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
```

### Operating System

macOs bigsur

### Selenium version

openjdk 13.0.1 2019-10-15 OpenJDK Runtime Environment (build 13.0.1+9) OpenJDK 64-Bit Server VM (build 13.0.1+9, mixed mode, sharing)

### What are the browser(s) and version(s) where you see this issue?

Appium 1.22.1 - iOS app

### What are the browser driver(s) and version(s) where you see this issue?

Appium 1.22.1 - iOS app

### Are you using Selenium Grid?

Selenium Grid 4.1.0

**Comment by titusfortner** [comment]
I'm not sure I understand your setup, but I'm not really familiar with advanced grid usage. Do you have an appium server set as a node of the grid? 

Since accessibility id is not a w3c supported locator strategy, I would not expect it to work with the Selenium Grid necessarily. @diemol do you know how previous grid handled this?

**Comment by kingages** [comment]
Yes, you are right. We have server with selenium grid 4 and server with appium 1.22.1 is connected as node. This setup worked with selenium grid 3.

**Comment by diemol** [comment]
The problem seems to be that `ACCESSIBILITY_ID` is not a W3C WebDriver locator, so it makes sense Grid 4 rejects it. What version of the Python Appium bindings are you using? Can you share the Python code being used to get this error?

**Comment by kingages** [comment]
Python 3.7 is used
Robot code:
```*** Keywords ***
On Page
    Landscape
    AppiumLibrary.Wait Until Element Is Visible  accessibility_id=login_title
```

Python code from appium library:
````   
def wait_until_element_is_visible(self, locator, timeout=None, error=None):
        def check_visibility():
            visible = self._is_visible(locator)
            if visible:
                return
            elif visible is None:
                return error or "Element locator '%s' did not match any elements after %s" % (locator, self._format_timeout(timeout))
            else:
                return error or "Element '%s' was not visible in %s" % (locator, self._format_timeout(timeout))
        self._wait_until_no_error(timeout, check_visibility)

**Comment by andriiTarasov** [comment]
@kingages 
You can use workaround for searching element by xpath for reading element with unique attribute name like

<code>
//[@name='login_title'][count(//[@name='login_title']) = 1]
</code>

**Comment by kingages** [comment]
Thank you for your idea but xpath are slower. However I was able to find in [appium documentation](https://github.com/appium/appium-xcuitest-driver#element-location) that it is possible use id locator instead of accessibility_id. ID locator works.

**Comment by diemol** [comment]
Thank you for sharing the solution, @kingages.
